### ISSUE TITLE
Function argument count is inconsistent when re-running a program.

### ISSUE THREAD

**Issue by ergthbeb** [KSP-KOS/KOS#1131]
I'm trying to implement a library of common methods that I plan to use in many programs.
I use a boot file to print craft info to the terminal.  I then planed to have the user manually run the Launch program via terminal input. That's where the problem is.

If one program runs my library program, the user cannot (through the terminal) run another program that runs my library program without issues.  When the launch program tries to call any function that has more than 0 arguments, the last argument is somehow forgotten, and an error that too many parameters are passed. (the last parameter can be omitted to get rid of the error, and make the code run, but it's missing argument info, and so does not print correctly.)

If I have the boot code go right into the launch script it seems to work, but I wanted that feeling that the user still decided to launch - not the program.

TL:DR It seems I cannot run a program via terminal if both that program and the boot file reference a common library program.

**Comment by Dunbaratu** [comment]
Make each program run the libraries it depends on, self contained.

Each RUN from the terminal spawns a new program context that gets new program code from scratch without access to the previous ones.  Inside a program, RUN will load into the SAME program code context, but from the terminal it will start a new one.

Docs reference covering it:  http://ksp-kos.github.io/KOS_DOC/language/user_functions.html#interpreter-functions

I realize the docs are not explicit about how functions in memory are wiped as soon as a terminal program run is over, but they are.  The docs cover the case where someone tried to run a program from the terminal and then use its functions from the terminal, and says that doesn't work, but it doesn't mention that this is because those functions are now gone (it should) and therefore running a second program after that won't let you get at them either.

Basically, the RUN from the terminal is slightly different from the RUN inside a program.  From the terminal, it's more like running an executable program at a shell prompt.  From inside a program, it's a bit more like doing a C #include statement, but one in which it only parses it the first time (only matters when you run the same program again and again in a loop from another program, which used to be how people had to do "functions" before proper functions got added), and thereafter just uses the preparsed one.

If lib_a.ks is a library file with important things in it that myprog.ks needs, and so is lib_b.ks, you can include them inside 'myprog.ks' by just doing ``run lib_a.  run lib_b.`` from inside 'myprog.ks.'

Make every top-level program that is intended to be launched from the interactive terminal run its own libraries, rather than rely on the user having run them beforehand for you.

**Comment by ergthbeb** [comment]
I really appreciate how thorough your responses are! My system works exactly as you have explained, but the problem persisted... After a while I realized that my function was the problem!
I've made a line-by-line printout inspired by Alien, so i made my own function "PrintAt" (you already know where I'm going now...) to include a delay parameter.
In the boot file, that was fine as a 'local' function, but when running via console that name appears reserved.
The same code was allowed at boot time, but after that it couldn't overload the "PrintAt" function so it took exception.
